### ISSUE TITLE
[$500] Web - The edit message is dismissed after getting hidden under the avatar

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#31165]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.3.97.0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Open any report
2. Chat a message
3. Go reply in thread
4. Click edit parent message
5. Click composer and send many messages (about > 50 messages)

## Expected Result:
The 'Edit Message' feature shouldn't be dismissed when it gets hidden under the avatar

## Actual Result:
The edit message is dismissed after getting hidden under the avatar

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] Windows: Chrome
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/f462def4-b390-4f53-af00-1c3f01ca4872

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01500218b2287a89b0</li>
        <li>Upwork Job ID: 1722777089310052352</li>
        <li>Last Price Increase: 2023-11-17</li>
    </ul>
</details>

**Comment by wlegolas** [comment]
Hi @lanitochka17 

I wasn't able to reproduce this issue using the latest version `v1.3.97-6`.


https://github.com/Expensify/App/assets/698363/13fd8648-59ba-460d-bd37-2fad9981af3f

**Comment by chamysoftware** [comment]
I can confirm I couldn't reproduce it either

**Comment by getusha** [comment]
I don't seem to reproduce this i think we can close this one
@conorpendergrast @lanitochka17

**Comment by barros001** [comment]
The issue here is not really that the edit is dismissed. If you watch closely the original video posted, the message being edited started as `Message 6` but once it's dismissed it reads `Message 62`. What's happening here is that once the edit goes out the screen, every time you send a new message, focus switches back to the edit. If you are sending a bunch of messages too fast you won't really notice and you will end up actually editing the message and hitting enter, which will save the edit. I was able to reproduce it locally.

So the problem here is not that it dismisses, but rather that focus is shifting back from new message composer to edit composer. 

This is a regression caused by https://github.com/Expensify/App/commit/657b79ebb192b5ca2d099f7bbe0d93d14d444bf3

It focus on mount and looks like the component unmounts and re-mounts once its hidden off the screen.

This has already been fixed by https://github.com/Expensify/App/commit/9f8248a85726a62fc2e154fb1619af4d43e0e1d0
